client/src/controllers/SwapController.ts

```diff
@@ -0,0 +1,239 @@
+  HTMLFormElement | HTMLInputElement
+> {
+  static targets = ['input'];
+
+  static values = {
+    icon: { default: '', type: String },
+    loading: { default: false, type: Boolean },
+    src: { default: '', type: String },
+    target: { default: '#results', type: String },
+    wait: { default: 200, type: Number },
+  };
+
+  declare readonly hasInputTarget: boolean;
+  declare readonly hasTargetValue: boolean;
+  declare readonly hasUrlValue: boolean;
+  declare readonly inputTarget: HTMLInputElement;
+
+  declare iconValue: string;
+  declare loadingValue: boolean;
+  declare srcValue: string;
+  declare targetValue: string;
+  declare waitValue: number;
+
+  /** Allow cancelling of in flight async request if disconnected */
+  abortController?: AbortController;
+  /** The related icon element to attach the spinner to */
+  iconElement?: SVGUseElement | null;
+  /** Element that receives the fetch result HTML output */
+  targetElement?: HTMLElement;
+  /** Debounced function to fetch results & patch into the DOM */
+  update?: { (...args: any[]): void; cancel(): void };
+
+  connect() {
+    // set up values that may be set with the global header approach
+    const formContainer = this.hasInputTarget
+      ? this.inputTarget.form
+      : this.element;
+    const { targetOutput, url: src } = getGlobalHeaderSearchOptions();
+    this.srcValue =
+      src || this.srcValue || formContainer?.getAttribute('action') || '';
+    this.targetValue = targetOutput || this.targetValue;
+    this.targetElement = this.getTarget(this.targetValue);
+
+    // set up icons
+    this.iconElement = null;
+    const iconContainer = (
+      this.hasInputTarget ? this.inputTarget : this.element
+    ).parentElement;
+
+    this.iconElement = iconContainer?.querySelector('use') || null;
+    this.iconValue = this.iconElement?.getAttribute('href') || '';
+
+    // set up initial loading state (if set originally in the HTML)
+    this.loadingValue = false;
+
+    // set up debounced update method
+    this.update = debounce(this.swapLocation.bind(this), this.waitValue);
+  }
+
+  getTarget(targetValue = this.targetValue) {
+    const targeElement = document.querySelector(targetValue);
+
+    const foundTarget = targeElement && targeElement instanceof HTMLElement;
+    const hasValidUrlValue = !!this.srcValue;
+
+    if (!foundTarget || !hasValidUrlValue) {
+      throw new Error(
+        [
+          !hasValidUrlValue && 'Cannot find valid src URL value',
+          !foundTarget &&
+            `Cannot find valid target element at "${targetValue}"`,
+        ]
+          .filter(Boolean)
+          .join(', '),
+      );
+    }
+
+    return targeElement as HTMLElement;
+  }
+
+  /**
+   * Toggle the visual spinner icon if available and ensure content about
+   * to be replaced is flagged as busy.
+   */
+  loadingValueChanged(isLoading: boolean) {
+    if (isLoading) {
+      this.targetElement?.setAttribute('aria-busy', 'true');
```

Comment:
I couldn't get this to be read by VoiceOver, but I might have used/configured it incorrectly

Reply:
I can do another smoke test for the aria-busy, I do recall (been a while) I had to slow down the voice read rate but I did validate that the screen reader correctly read out that the content had been updated.

Reply:
Maybe also slow down your network when running locally, aria-busy will only trigger for <300ms when doing a local API request.

Reply:
Nice digging, thanks. Keen to maybe re-look at this in the future though.